File: PHP/1. Two Sum.php
<?php
// 1. Two Sum
// Level: Easy
// https://leetcode.com/problems/two-sum/

class Solution
{
  function twoSum($nums, $target)
  {
    $map = [];
    for ($i = 0; $i < count($nums); $i++) {
      if (array_key_exists($nums[$i], $map))
        return [
          $i, $map[$nums[$i]]
        ];
      $map[$target - $nums[$i]]  = $i;
    }
    return [-1, -1];
  }
}

File: CSharp/1. Two Sum.cs
// 1. Two Sum
// Level: Easy
// https://leetcode.com/problems/two-sum/

public class Solution
{
  public int[] TwoSum(int[] nums, int target)
  {
    var map = new Dictionary<int, int>();
    for (int i = 0; i < nums.Length; i++)
    {
      if (map.ContainsKey(nums[i]))
        return new int[] { i, map[nums[i]] };
      map[target - nums[i]] = i;
    }

    return new int[] { -1, -1 };
  }
}
